crates/datasources/src/object_store/http.rs

```diff
@@ -140,3 +168,193 @@ impl ObjStoreAccess for HttpStoreAccess {
             .await?)
     }
 }
+
+#[derive(Debug)]
+struct SimpleHttpStore {
+    url: Url,
+    // Used when content length available.
+    obj_store: HttpStore,
+}
+
+impl SimpleHttpStore {
+    async fn simple_get_req(&self, location: ObjectStorePath) -> Result<GetResult> {
+        let res = reqwest::get(self.url.clone()).await?;
+        if !res.status().is_success() {
+            return Err(ObjectStoreSourceError::InvalidHttpStatus(format!(
+                "getting data for '{}' resulted in error status: {}",
+                self.url,
+                res.status(),
+            )));
+        }
+
+        // TODO: Maybe write the byte stream to file?
+        // Would only be useful when the returned bytes are too many.
+        let contents = res.bytes().await?;
+
+        let size = contents.len();
+
+        let stream = async_stream::stream! {
+            let res: Result<Bytes, object_store::Error> = Ok(contents);
+            yield res;
+        };
+
+        let payload = GetResultPayload::Stream(Box::pin(stream));
```

Comment:
my concern here is less that we should write it to a file (or not) but that we actually/unintentionally end up reading the whole stream into memory here... That's the main risk in my mind...